### ISSUE TITLE
推导交叉熵损失函数？

### ISSUE THREAD

**Issue by Sogrey** [Sogrey/AI-QA#203]
交叉熵损失函数（Cross-Entropy Loss）是机器学习和深度学习中常用的损失函数，尤其在分类任务中表现突出。以下是关于交叉熵损失函数的详细解释：

------
### 📌 **1. 基本定义**
交叉熵损失函数用于衡量模型预测概率分布与真实标签分布之间的差异。其公式如下：
$$ L = -\sum_{i=1}^{n} y_i \log(\hat{y}_i) $$
其中，$ y_i $ 是真实标签（0或1），$ \hat{y}_i $ 是模型预测的概率。

------
### 📌 **2. 适用场景**
交叉熵损失函数广泛应用于以下场景：
- **二分类任务**：如垃圾邮件检测，使用 sigmoid 函数输出概率。
- **多分类任务**：如图像分类，使用 softmax 函数输出概率分布。

------
### 📌 **3. 优点**
- **直观性**：直接反映预测概率与真实标签的差异。
- **可导性**：便于使用梯度下降等优化算法进行模型训练。
- **灵活性**：适用于不同类型的分类任务（二分类或多分类）。

------
### 📌 **4. 与其他损失函数的对比**
| **损失函数** | **适用场景** | **特点** |
|--------------|--------------|----------|
| **均方误差（MSE）** | 回归任务 | 适用于连续值预测，对异常值敏感 |
| **交叉熵损失函数** | 分类任务 | 适用于离散标签，能有效优化分类模型 |

------
### 📌 **5. 示例代码（PyTorch）**
```python
import torch
import torch.nn as nn

# 定义交叉熵损失函数
criterion = nn.CrossEntropyLoss()

# 假设的预测值和真实标签
predictions = torch.tensor([[0.1, 0.9], [0.2, 0.8]], requires_grad=True)
labels = torch.tensor([1, 0])

# 计算损失
loss = criterion(predictions, labels)
print(loss.item())
```

------
### 📌 **6. 总结**
交叉熵损失函数是分类任务中不可或缺的一部分，其直观性和可导性使其成为深度学习模型训练的首选损失函数之一。在实际应用中，根据任务类型选择合适的损失函数能够显著提升模型性能。

> 💡 **提示**：在使用交叉熵损失函数时，确保输出层的激活函数与损失函数相匹配（如 softmax 用于多分类，sigmoid 用于二分类）。

======

来自n8n
